test/framework/src/main/java/org/elasticsearch/logsdb/datageneration/arbitrary/RandomBasedArbitrary.java

```diff
@@ -51,6 +52,32 @@ public long longValue() {
 
     @Override
     public String stringValue(int lengthLowerBound, int lengthUpperBound) {
-        return randomAlphaOfLengthBetween(lengthLowerBound, lengthLowerBound);
+        return randomAlphaOfLengthBetween(lengthLowerBound, lengthUpperBound);
+    }
+
+    @Override
+    public boolean generateNullValue() {
+        // Using a static 10% chance, this is just a chosen value that can be tweaked.
+        return randomDouble() < 0.1;
+    }
+
+    @Override
+    public boolean generateArrayOfValues() {
```

Comment:
The `ArrayGenerator` can take a generic `Generator` as a constructor argument to which we can delegate the actual generation of array values. This could then be a `NullValueGenerator` another `ArrayGenerator` (to generate nested arrays),...That would also allow us to generate malformed values.